go-controller/pkg/testing/mock_ovn.go

```diff
@@ -48,6 +49,9 @@ type MockOVNClient struct {
 	// error injection
 	// keys are of the form: Table:Name:FieldType
 	errorCache map[string]error
+	// Mutex synchronizing the cache when running the
+	// tests with the race detector "on"
+	errorCacheMutex sync.Mutex
```

Comment:
`cache` is not thread-safe either. While we may not _currently_ be using it from multiple threads, you might as well just add a single top-level `sync.Mutex` to `MockOVNClient` and use it for all methods.

Also, you don't need to reference the race detector specifically; if we are accessing the fields from multiple goroutines, then that's a bug whether or not we are running the race detector. The race detector is just _finding_ the bug. It's not the cause of the bug.